bytecode/concrete.py

```diff
@@ -439,9 +439,9 @@ def __init__(self, code):
     def add_const(self, value):
         key = const_key(value)
         if key in self.consts:
-            return self.consts[key]
+            return self.consts[key][0]
         index = len(self.consts)
-        self.consts[key] = index
+        self.consts[key] = (index, value)
```

Comment:
Alternatively, you could append `value` to the `self.consts_list` list.

Reply:
Not sure what you are referring to here. There is no `self.consts_list`.

Reply:
Currently `self.consts` is a dict mapping the const key to the const index. You make it mapping to the pair of the const index and the const value. Alternatively you could introduce a list `self.consts_list` and append the const value to it. The code in `to_concrete_bytecode()` for converting `self.consts` to a list of constants can be eliminated, since `self.consts_list` is such a list.